# SciKit-Learn Primer

Scikit-Learn is a open-source machine learning library for Python that offers a variety of classification, regression and clustering algorithms.

In this notebook we'll perform a fairly simple classification exercise with scikit-learn. 

In this notebook we'll be building a machine learning model that will be able to tell which message is a spam and which one is ham.

# Perform Import  and Load Data
For this exercise we'll be using the **SMSSpamCollection** dataset from [UCI datasets](https://archive.ics.uci.edu/ml/datasets/SMS+Spam+Collection) that contains more than 5 thousand SMS phone messages.<br>You can check out the [**sms_readme**](../TextFiles/sms_readme.txt) file for more info.

The file is a [tab-separated-values](https://en.wikipedia.org/wiki/Tab-separated_values) (tsv) file with four columns:
> **label** - every message is labeled as either ***ham*** or ***spam***<br>
> **message** - the message itself<br>
> **length** - the number of characters in each message<br>
> **punct** - the number of punctuation characters in each message

In [2]:
# Import tools 
import numpy as np
import pandas as pd

C:\Users\DELL\Anaconda3\envs\nlp_course\lib\importlib\_bootstrap.py:219: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88, got 80
  return f(*args, **kwds)
C:\Users\DELL\Anaconda3\envs\nlp_course\lib\importlib\_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216, got 192
  return f(*args, **kwds)
C:\Users\DELL\Anaconda3\envs\nlp_course\lib\importlib\_bootstrap.py:219: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88, got 80
  return f(*args, **kwds)
C:\Users\DELL\Anaconda3\envs\nlp_course\lib\importlib\_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216, got 192
  return f(*args, **kwds)


In [3]:
df=pd.read_csv("../TextFiles/smsspamcollection.tsv",sep="\t")
df.head()

label                                            message  length  punct
0   ham  Go until jurong point, crazy.. Available only ...     111      9
1   ham                      Ok lar... Joking wif u oni...      29      6
2  spam  Free entry in 2 a wkly comp to win FA Cup fina...     155      6
3   ham  U dun say so early hor... U c already then say...      49      6
4   ham  Nah I don't think he goes to usf, he lives aro...      61      2

In [4]:
len(df)

5572

In [6]:
# Let's visualize our data to understand our data
len(df[df['label']=='spam'])

747

<color=blue>So our data have 747 spam messages and rest are ham messages